### ISSUE TITLE
Portal focus, magic doors and non-euclidean spaces

### ISSUE THREAD

**Issue by angeltxilon** [Azanor/thaumcraft-suggestions#131]
We currently have the wand focus "portable hole", but I think we could go further.

- For example, I thought in the addition of a **focus "portal".** 

_How it works?_
Click in the air of an specific zone using the wand with the focus "portal" (linking step).
Next, if you are far and you want to go to the previous zone, click in the air again with using the wand with such focus, and a portal showing the destination zone will be created (portal opening step).
This portal will disappear when the player click again in the air using the wand with the focus "portal", and then the wand will stop of be linked to such zone, being able to be used again for create portals to others selected zones (closing/unlinking step).

The link (first step) and unlink (third step) doesn't consume vis, however, the creation of portals (second step) will consume big quantities of ordor and perditio vis.
It doesn't work between dimensions.

_Illustration_ about the concept: http://i.imgur.com/YfBhB4o.jpg
(I know that this is an image of the arcane portal of thaumcraft 2, but it still being a good image for illustrate the concept. Obviously, the wand and focus "portal" will work in a very different way."

---

- Another idea is to add **thaumic doors**. 
> What is better than a door and better than a train? Yes, the thaumic door is all what you needs.
>Open this door will not open access to you to go to the other side of the door, but will open access to go more far... more, more far.

Thaumic doors would be an special type of door based in magnific wood, silverwood or brass (depending of the likes of the thaumaturge in question). For example, magnific door could have an rustic traditional texture, silverwoord door an fairy aspect and brass door could be of steampunk style.

These would be created in the arcane crafting table with a modified "door pattern" (ie, using special items like ender pearls) and needing vis in the process of crafting.

_How it works?_
These doors, next of its placement, will work as a common door. For be used for large distances translations, a pair of these (of the same type both) must be linked.
Maybe a new special item should be needed for such action, or, alternatively, the portal focus of before.

Once two doors are linked: 
When you open one door, the another door must open at the same time.
Once you opened one linked door, the landscape through this will correspond to the landscape of the other door and not to the landscape behind.
This is, these two doors acts like a kind of portal, a very subtle and elegant portal.

It should be possible also to create linked double thaumic doors.

_Illustrations:_
http://i.imgur.com/gkecUFB.png
http://i.imgur.com/dKbk3qB.png
(imagine that these common doors are actually special thaumic doors).

_Note_: these doors can't be linked between dimensions, ie, only can be linked to another door of the same dimension.

However, and additionally, it could be added special doors (may be based in void metal?) able to connect different places of different dimensions.

---

- Finally I invite all you to raise the possibilities of adding non-Euclidean spaces based on the rendering of portal images.
Imagine non-Euclidean structures and corridors in temples or outer lands, non-euclidean labyrinths in the dimension of the end, etc.
The possibilities could be endless, with the only cost of adding an instant teleportation and pre-rendering system of images in invisible special portal blocks. This, although hard, should be possible to implement to the game.

For those of you who do not know what I'm talking about, I recommend watching videos or trailers about non-Euclidean space games (portal, fragments of euclid, antichamber, manifold garden, etc). 
These strange effects of space shown, (like, rooms bigger inside than outside, fall infinitely seeing all the time the same landscape, physically impossible structures like roundabouts of more than 360 degrees, etc) is what is called "non-Euclidean space" in video games.

**Comment by webwraith** [comment]
I agree with the idea of the distance doors, possibly being a multi-block structure, and requiring some sort of outer frame, to keep it in theme, although I'm not sure that Minecraft would be able to handle non-euclidean spaces, beyond, perhaps, TARDIS-like spaces, and even then, I'm dubious. Everything you've said sounds fantastic, I'm just worried that the game, regardless of whether it's been modded or not, could not do what you're suggesting. 

Teleports would work, doors showing the other side could work, but bear in mind that the game would then have to load twice as many chunks into memory, whenever the door is in a loaded chunk, just so you can see through it, and even walk through it, for that matter.

**Comment by Vitzgraco** [comment]
The ideas about portal focus and distance doors seems very interesting, I would like to see these in game.

Edit: I hope you do not mind linking this suggestion with a comment on my idea.

**Comment by angeltxilon** [comment]
No problem, vitzgraco.
